charts/templates/statefulset.yaml

```diff
@@ -0,0 +1,188 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "marklogic.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "marklogic.labels" . | nindent 4 }}
+spec:
+  serviceName: {{ include "marklogic.headlessServiceName" . }}
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "marklogic.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "marklogic.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "marklogic.serviceAccountName" . }}
+      containers:
+        - name: marklogic-server
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy}}
+          volumeMounts:
+            - name: datadir
+              mountPath: {{ .Values.persistence.mountPath }}
+          env:
+            - name: ML_ADMIN_USER
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "marklogic.fullname" . }}
+                  key: username
+            - name: ML_ADMIN_PASS
+              valueFrom:
+                  secretKeyRef:
+                    name: {{ include "marklogic.fullname" . }}
+                    key: password
+          envFrom:
+            - configMapRef:
+                name: {{ include "marklogic.fullname" . }}
+          command:
+            - bash
+            - '-c'
+            - |
+              echo '### Begin ML Container Config ###'
+
+              # Set Hostname to the value of hostname command to /etc/marklogic.conf
+              HOST_FQDN="$(hostname).$ML_FQDN_SUFFIX"
+              echo "export MARKLOGIC_HOSTNAME=\"$HOST_FQDN\"" | sudo tee /etc/marklogic.conf
+
+              cd ~
+
+              AUTH_CURL="curl --anyauth --user $ML_ADMIN_USER:$ML_ADMIN_PASS -m 20 -s "
+
+              # Start MarkLogic
+              sudo service MarkLogic start
+              sleep 5s
+
+              # Initialize and Setup Admin User
+              echo '### Initialize and Setup Admin User ###'
+              ML_STATUS_CODE=`curl -s -o /dev/null --write-out %{response_code}  http://localhost:8001/admin/v1/init`
+              if [ "$ML_STATUS_CODE" == "401" ]; then
+                echo "Server is already configured." 
+              else
+                curl -X POST -d "" http://$HOSTNAME:8001/admin/v1/init
+                sleep 5s
+
+                curl -X POST -H "Content-type: application/x-www-form-urlencoded" \
+                    --data "admin-username=$ML_ADMIN_USER" --data "admin-password=$ML_ADMIN_PASS" \
+                    --data "realm=public" \
+                    http://$HOSTNAME:8001/admin/v1/instance-admin
+                sleep 10s
+
+                # Turn on XDQP encryption via CMA
```

Comment:
Please comment this out for ea1 as I do not want there to be any confusion about performance in k8s vs non-k8s. If a customer uses XDQP ssl, they can turn it on.

Reply:
I've comment out this for EA1